libcxx/include/__flat_set/flat_set.h

```diff
@@ -0,0 +1,845 @@
+      // C++23 Sequence Container should have insert_range member function
+      // Note that not all Sequence Containers provide append_range.
+      __keys_.insert_range(__keys_.end(), std::forward<_Range>(__rng));
+    } else if constexpr (ranges::common_range<_Range>) {
+      __keys_.insert(__keys_.end(), ranges::begin(__rng), ranges::end(__rng));
+    } else {
+      for (auto&& __x : __rng) {
+        __keys_.insert(__keys_.end(), std::forward<decltype(__x)>(__x));
+      }
+    }
+  }
+
+  template <bool _WasSorted, class... _Args>
+  _LIBCPP_HIDE_FROM_ABI void __append_sort_merge_unique(_Args&&... __args) {
+    auto __on_failure    = std::__make_exception_guard([&]() noexcept { clear() /* noexcept */; });
+    size_type __old_size = size();
+    __append(std::forward<_Args>(__args)...);
+    if (size() != __old_size) {
+      if constexpr (!_WasSorted) {
+        ranges::sort(__keys_.begin() + __old_size, __keys_.end(), __compare_);
+      } else {
+        _LIBCPP_ASSERT_SEMANTIC_REQUIREMENT(__is_sorted_and_unique(__keys_ | ranges::views::drop(__old_size)),
+                                            "Either the key container is not sorted or it contains duplicates");
+      }
+      ranges::inplace_merge(__keys_.begin(), __keys_.begin() + __old_size, __keys_.end(), __compare_);
+
+      auto __dup_start = ranges::unique(__keys_, __key_equiv(__compare_)).begin();
+      __keys_.erase(__dup_start, __keys_.end());
+    }
+    __on_failure.__complete();
+  }
+
+  template <class _Self, class _Kp>
+  _LIBCPP_HIDE_FROM_ABI static auto __find_impl(_Self&& __self, const _Kp& __key) {
+    auto __it   = __self.lower_bound(__key);
+    auto __last = __self.end();
+    if (__it == __last || __self.__compare_(__key, *__it)) {
+      return __last;
+    }
+    return __it;
+  }
+
+  template <class _Self, class _Kp>
+  _LIBCPP_HIDE_FROM_ABI static auto __equal_range_impl(_Self&& __self, const _Kp& __key) {
+    using __iter = _If<is_const_v<__libcpp_remove_reference_t<_Self>>, const_iterator, iterator>;
+    auto __it    = ranges::lower_bound(__self.__keys_, __key, __self.__compare_);
+    auto __last  = __self.__keys_.end();
+    if (__it == __last || __self.__compare_(__key, *__it)) {
+      return std::make_pair(__iter(__it), __iter(__it));
+    }
+    return std::make_pair(__iter(__it), __iter(std::next(__it)));
+  }
+
+  template <class _KeyArg>
+  _LIBCPP_HIDE_FROM_ABI iterator __emplace_exact_pos(const_iterator __it, _KeyArg&& __key) {
+    auto __on_failure = std::__make_exception_guard([&]() noexcept {
+      if constexpr (!__container_traits<_KeyContainer>::__emplacement_has_strong_exception_safety_guarantee) {
+        clear() /* noexcept */;
+      }
+    });
+    auto __key_it     = __keys_.emplace(__it.__base(), std::forward<_KeyArg>(__key));
+    __on_failure.__complete();
+    return iterator(std::move(__key_it));
+  }
+
+  template <class _Kp>
+  _LIBCPP_HIDE_FROM_ABI pair<iterator, bool> __try_emplace(_Kp&& __key) {
```

Comment:
```suggestion
  _LIBCPP_HIDE_FROM_ABI pair<iterator, bool> __emplace(_Kp&& __key) {
```

Since there is no `flat_set::try_emplace` function, it feels weird to have `__try_emplace`.